lib/Runtime/Library/TypedArray.cpp

```diff
@@ -2252,40 +2213,59 @@ namespace Js
         Assert(!(callInfo.Flags & CallFlags_New));
         CHAKRATEL_LANGSTATS_INC_BUILTINCOUNT(TypedArray_Prototype_sort);
 
+        // Throw if not a valid typed array
+        // This step is per spec and allows us to optimise significantly below
         TypedArrayBase* typedArrayBase = ValidateTypedArray(args, scriptContext, _u("[TypedArray].prototype.sort"));
         uint32 length = typedArrayBase->GetLength();
 
-        // If TypedArray has no length, we don't have any work to do.
+        // Early return if length is 0
         if (length == 0)
         {
             return typedArrayBase;
         }
 
         RecyclableObject* compareFn = nullptr;
 
+        // Spec requires us to throw if comparison function is neither undefined nor callable
         if (args.Info.Count > 1 && !JavascriptOperators::IsUndefined(args[1]))
         {
             if (!JavascriptConversion::IsCallable(args[1]))
             {
                 JavascriptError::ThrowTypeError(scriptContext, JSERR_FunctionArgument_NeedFunction, _u("[TypedArray].prototype.sort"));
             }
 
-            compareFn = VarTo<RecyclableObject>(args[1]);
+            compareFn = UnsafeVarTo<RecyclableObject>(args[1]);
         }
```

Comment:
Just noticed that this is wrong - but will hold off on fixing until review is complete.
Per spec `Uint8Array.from([]).sort(5)` or equivalent should throw - sort with invalid comparison method.

With current logic if the length is 0 we don't throw. Steps to be re-ordered to match spec.